### DISCUSSION TITLE
What is the difference between `pushed` and `updated` directions in the list repos for Org API

### DISCUSSION THREAD

**Discussion by GovindarajanNagarajan-TomTom** [community/community#112102] [Apps, API and Webhooks]
### Select Topic Area

Product Feedback

### Body

The documentation on the [List Repos for Org](https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#list-organization-repositories) isn't very clear as to what the difference between a `pushed` and a `updated` sort direction is. 
Is `pushed` implying a commit/tag  that has been pushed and `updated` meaning that some setting or Repo description was updated ? 

Also, may we request a feature in the Github API for an API that can list only repos that have changes in a particular time range ? 

The API will be used to give us insights into how our teams productivity and usage and will help us build a lot of custom analytics to improve the Dev Ex

**Comment by ChadwickBroski** [comment]
# Difference Between `pushed` and `updated` in GitHub Org Repos API

In the GitHub REST API for listing organization repositories:

```http
GET /orgs/{org}/repos
```

The `sort` parameter can use:

- `pushed`
- `updated`

These mean different things.

---

# `pushed`

This refers to the last time code was pushed to the repository.

It changes when:
- commits are pushed
- branches are updated
- tags are pushed

It is based only on Git activity.

Example:

- Someone pushes a new commit  
  → `pushed_at` changes

- Someone edits the repo description  
  → `pushed_at` does NOT change

---

# `updated`

This refers to the last time *anything* in the repository changed.

It changes for:
- code pushes
- issues
- pull requests
- releases
- wiki edits
- repo settings
- description/topic changes
- some discussion activity

It is broader than `pushed`.

Example:

- Someone edits the repository description  
  → `updated_at` changes

- No code was pushed  
  → `pushed_at` stays the same

---

# Sorting Difference

## `sort=pushed`

Sorts repositories by latest code push activity.

Good for:
- actively coded repositories
- recent commits

Example:

```http
GET /orgs/myorg/repos?sort=pushed
```

---

## `sort=updated`

Sorts repositories by latest overall activity.

Good for:
- generally active repositories
- repos with issue/PR/discussion activity

Example:

```http
GET /orgs/myorg/repos?sort=updated
```

---

# Simple Summary

| Sort Type | Tracks |
|---|---|
| `pushed` | Code pushes only |
| `updated` | Any repository activity |

**Comment by P-r-e-m-i-u-m** [comment]
Yo @GovindarajanNagarajan-TomTom — your hypothesis is spot on.

pushed: You nailed it. This is strictly the timestamp of the last commit pushed to the default branch. It's the best metric for actual code activity.

updated: This is broader. It includes metadata changes like updating the description, changing settings, adding collaborators, or even someone starring the repo. It tracks “repo object” activity, not just coding.

For your analytics feature:
You don't actually need a new API endpoint to do this. You can use the Search API with date queries.

Try this:

q=org:YourOrg pushed:2024-03-01..2024-03-11

It filters repos based on activity ranges efficiently without you having to list and filter them manually. It's way faster for analytics.
